x-pack/plugins/canvas/public/components/app/index.js

```diff
@@ -46,13 +44,11 @@ const types = {
 
 const mapDispatchToProps = dispatch => ({
   // TODO: the correct socket path should come from upstream, using the constant here is not ideal
-  setAppReady: basePath => async () => {
+  setAppReady: () => async () => {
```

Comment:
These changes must be validated by the Canvas team. I think canvas shows it's loading state while `setAppReady` hasn't resolved yet, but moving these functions out there, doesn't make this function wait anymore on the interpreter to be ready for it to be resolved.

Maybe we need a callback on the interpreter to know when it's ready (or basically have two separate interpreters?)

Reply:
we wait for the browser registries to be loaded, and that happens after we initialize the interpreter